src/main/java/org/springframework/data/redis/core/convert/MappingRedisConverter.java

```diff
@@ -317,7 +317,7 @@ protected Object readProperty(String path, RedisData source, RedisPersistentProp
 			return null;
 		}
 
-		if (persistentProperty.isIdProperty() && ObjectUtils.isEmpty(path.isEmpty())) {
+		if (persistentProperty.isIdProperty() && !StringUtils.hasText(path.isEmpty())) {
```

Comment:
This is still wrong as we want to check whether `path` is empty, not whether `isEmpty` returns an empty result.